lib/tlscontext.c

```diff
@@ -326,6 +346,13 @@ tls_session_new(SSL *ssl, TLSContext *ctx)
 
   SSL_set_info_callback(ssl, tls_session_info_callback);
 
+  if (!_set_sni_in_client_mode(self))
+    {
+      tls_context_unref(self->ctx);
+      g_free(self);
+      return NULL;
```

Comment:
If this fails, a `SSL` object will be leaked every `time-reopen()` seconds.

```
324,918 (244,920 direct, 79,998 indirect) bytes in 39 blocks are definitely lost in loss record 97 of 97
   at 0x483877F: malloc (vg_replace_malloc.c:299)
   by 0x4C5F59A: CRYPTO_zalloc (in /usr/lib/libcrypto.so.1.1)
   by 0x4A92485: SSL_new (in /usr/lib/libssl.so.1.1)
   by 0x48A9551: tls_context_setup_session (tlscontext.c:771)
   by 0x5D71C8A: _construct_tls_transport (transport-mapper-inet.c:100)
   by 0x5D71DFC: transport_mapper_inet_construct_log_transport (transport-mapper-inet.c:145)
   by 0x5D6BFA2: transport_mapper_construct_log_transport (transport-mapper.h:83)
   by 0x5D6C896: afsocket_dd_construct_transport (afsocket-dest.c:213)
   by 0x5D6C9BC: afsocket_dd_connected (afsocket-dest.c:231)
   by 0x5D6CD8C: _afsocket_dd_connection_in_progress (afsocket-dest.c:290)
   by 0x499CCDF: ??? (in /usr/lib/libivykis.so.0.5.6)
   by 0x499E293: iv_main (in /usr/lib/libivykis.so.0.5.6)
```

Reply:
Thanks, fixed! :)